recipe/meta.yaml

```diff
@@ -1,39 +1,59 @@
 {% set name = "yapf" %}
-{% set version = "0.31.0" %}
-{% set sha256 = "e6330f89f024615b102f3140e50a1b864196c4d9d8e29ace4eaa234feec4c924" %}
+{% set version = "0.40.2" %}
+{% set sha256 = "8e0825803c488aa2cb118ed277a18d4617984608e75d10d5bffa5e24734eb022" %}
 
 package:
   name: {{ name }}
   version: {{ version }}
 
 source:
-  fn: {{ name }}-{{ version }}.tar.gz
   url: https://github.com/google/{{ name }}/archive/v{{ version }}.tar.gz
   sha256: {{ sha256 }}
 
 build:
   entry_points:
     - yapf = yapf:run_main
+    - yapf-diff = yapf_third_party.yapf_diff.yapf_diff:main
   number: 0
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
-  noarch: python
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
+  skipt: true # [py<37]
 
 requirements:
   host:
     - python
     - pip
+    - setuptools
+    - wheel
   run:
     - python
+    - importlib-metadata >=6.6.0
+    - platformdirs >=3.5.1
+    - tomli >=2.0.1
 
 test:
+  source_files:
+    - yapftests
+  imports:
+    - yapf
   commands:
+    - pip check
     - yapf --help
-    - yapf --version
+    - yapf-diff --help
+    - python -m unittest discover -p '*_test.py' yapftests/
+  requires:
+    - pip
 
 about:
   home: https://github.com/google/yapf
-  license: Apache 2.0
-  summary: A formatter for Python files
+  license: Apache-2.0 AND EPL-2.0
```

Comment:
Where does EPL come from?

Reply:
It should have been PSF from the thirdparty. I went too fast. I'll amend.